Improve DocumentsTab filter input

languy opened the pull request:
[Preview this branch](https://cosmos-explorer-preview.azurewebsites.net/pull/1998)
 * Always keep filter editable
 * Single "Apply filter"/"Cancel"
 * Fluent UI-based dropdown for saved and default filters with link to learn more. 
 * Clear button on the right clear filter
 * Cleaned up old style and code (for example don't focus with outline)
 * Input and dropdown navigation with tab for a11y compliance
 * The input and dropdown menu is a React component.

<img width="1052" alt="image" src="https://github.com/user-attachments/assets/a2977d32-4493-433e-86af-dcfba8831b73">

#### Behavior of the dropdown and keyboard navigation
* Dropdown will show when input gets focus
* Dropdown shows only when input has no text
* Clear button at the end of the input box becomes down button to reveal dropdown if input has no text. 
* Generally first dropdown item gets autofocus when dropdown opens (when input doesn't have focus)
* Arrow up/down move focus in the dropdown item. If last item, we focus back to the input.
* Tab or shift tab also work. If last item, tab goes to the Apply button.

sevoku commented:
This looks great, however I see 2 issues:
1. a11y: there is no way to enter the dropdown with suggestions/history using the keyboard, when entering the text entry the dropdown is shown, however I can't select anything using the arrow keys or tab into it (without the change I can use up/down arrow to select something)
2. when the entry is empty, the "X" button remains. I think it should change to a "Down Arrow" to reveal the history (and also might be the alternative entry point for keyboard accessibility):
![image](https://github.com/user-attachments/assets/e202c524-8ba2-4b2a-9946-56a3c2635411)

languy commented:
> This looks great, however I see 2 issues:
> 
> 1. a11y: there is no way to enter the dropdown with suggestions/history using the keyboard, when entering the text entry the dropdown is shown, however I can't select anything using the arrow keys or tab into it (without the change I can use up/down arrow to select something)
> 2. when the entry is empty, the "X" button remains. I think it should change to a "Down Arrow" to reveal the history (and also might be the alternative entry point for keyboard accessibility):
>    ![image](https://private-user-images.githubusercontent.com/951587/380904893-e202c524-8ba2-4b2a-9946-56a3c2635411.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzAzMDE1NjIsIm5iZiI6MTczMDMwMTI2MiwicGF0aCI6Ii85NTE1ODcvMzgwOTA0ODkzLWUyMDJjNTI0LThiYTItNGIyYS05OTQ2LTU2YTNjMjYzNTQxMS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDMwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAzMFQxNTE0MjJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05OWI4MTFiNzA0YjNmNjhiOWNkNmY4MTczYjMyMTI5ZmQ0NWY0ZWVkMmRhMzc2NWY4YzVhZDY4YzBlMzliN2I2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.MrkzcArCH_HHznzp4f4LpbponKz100mPWeM9uOftKbA)

Good points. Fixed and added more description about keyboard navigation.

sevoku reviewed (approved):
Thanks! Much better!
